integration_tests/juniper_tests/src/issue_398.rs

```diff
@@ -0,0 +1,67 @@
+// Original author of this test is <https://github.com/davidpdrsn>.
+use juniper::*;
+
+struct Query;
+
+#[juniper::object]
+impl Query {
+    fn users(executor: &Executor) -> Vec<User> {
+        // This doesn't cause a panic
+        executor.look_ahead();
+
+        vec![User {
+            country: Country { id: 1 },
+        }]
+    }
+}
+
+struct User {
+    country: Country,
+}
+
+#[juniper::object]
+impl User {
+    fn country(&self, executor: &Executor) -> &Country {
+        // This panics!
+        executor.look_ahead();
+
+        &self.country
+    }
+}
+
+struct Country {
+    id: i32,
+}
+
+#[juniper::object]
+impl Country {
+    fn id(&self) -> i32 {
+        self.id
+    }
+}
+
+type Schema = juniper::RootNode<'static, Query, EmptyMutation<()>>;
+
+#[test]
+fn it_works() {
```

Comment:
That's great, but a different name would still be good. :P

Reply:
😅